### ISSUE TITLE
Flow Distribution when encountering Equal Costs

### ISSUE THREAD

**Issue by VJander** [pywr/pywr#962]
I've developed an relatively complex model of water allocation that includes reservoirs, demand sites and hydropower plants, with streamflow gauges to represent the water rights. The problem is that, when encountering equal costs between nodes given a water deficit, the PYWR model distributes not with equality but with priority on filling the first node in the route. Is there any way that, when encountering deficit, the flow is assigned in equal parts to each of the demand/reservoir/hydroplant nodes?

This is the equivalent of creating an established priority, for example with the WEAP software, and allocating water in equal measure to each demand site in case of water deficit.

**Comment by jetuk** [comment]
> Is there any way that, when encountering deficit, the flow is assigned in equal parts to each of the demand/reservoir/hydroplant nodes?

I don't believe there is an easy way to do this because it's the nature of the simplex method to end up at a vertex (i.e. not split this between the nodes) when there are multiple optimal solutions. We try to make this deterministic by setting up the LP in the same way each time. However, as you have found the solution of where the deficit is placed is arbitrary. 

One option is to create a dedicated "deficit node" that is a supplier of last resort and connected to all demand nodes. This would have a high penalty cost. Any flow through this node would indicate a deficit somewhere in the model and could be interpreted as being equally distributed. However, that interpretation might be incorrect if the network can supply some nodes but not others.
